tests/sql-proxy.sh

```diff
@@ -0,0 +1,132 @@
+#!/bin/bash
+
+dir=$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )
+source "${dir}/helpers.bash"
+# dir might have been overwritten by helpers.bash
+dir=$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )
+
+TEST_NAME=$(get_filename_without_extension $0)
+LOGS_DIR="${dir}/cilium-files/${TEST_NAME}/logs"
+redirect_debug_logs ${LOGS_DIR}
+
+set -ex
+
+function cleanup {
+  monitor_stop
+  cilium policy delete --all 2> /dev/null || true
+  docker rm -f mysql-server mysql-client 2> /dev/null || true
+}
+
+function finish_test {
+  echo cleanup
+}
+
+trap finish_test EXIT
+
+SERVER_LABEL="mysql-server"
+CLIENT_LABEL="mysql-client"
+TAG="5.5"
+
+CLIENT_RUN="docker run --rm -t --net=cilium --name mysql-client -l mysql-client mysql:$TAG mysql -ucilium -pcilium --disable-ssl"
+cleanup
+logs_clear
+
+function proxy_init {
+  log "beginning proxy_init"
+  create_cilium_docker_network
+
+  docker run -dt --net=cilium --name mysql-server -l $SERVER_LABEL -e MYSQL_ROOT_PASSWORD=cilium --publish 6603:3306 mysql:$TAG --disable-ssl
+  wait_for_docker_ipv6_addr mysql-server
+
+  log "waiting for all 3 endpoints to get an identity"
+  while [ `cilium endpoint list -o jsonpath='{range [*]}{.status.identity.id}{"\n"}{end}' | grep '^[0-9]' | grep -v '^5$' | wc -l` -ne 3 ] ; do
```

Comment:
Are we assuming that there will be 3 endpoints after running MySQL? On my fresh Vagrant environment I end up having 4 endpoints just after running this test, so I end up having an infinite loop.

Maye we can wait for the mysql endpoint explicitly? `cilium endpoint list | grep mysql-server`?

Reply:
This script works for me after this change:

```suggestion
  while ! cilium endpoint list | grep mysql-server ; do
```

Reply:
Fixed, thanks!